java/source/flight.rst

```diff
@@ -0,0 +1,1699 @@
+.. _arrow-flight:
+
+============
+Arrow Flight
+============
+
+Recipes related to leveraging Arrow Flight protocol
+
+.. contents::
+
+Simple Service with Arrow Flight
+================================
+
+We are going to create: Flight Producer and Fligh Server:
+
+* InMemoryStore: A FlightProducer that hosts an in memory store of Arrow buffers. Used for integration testing.
+
+* ExampleFlightServer: An Example Flight Server that provides access to the InMemoryStore.
```

Comment:
I know the integration test has a separate server class, but IMO we don't need to create a server class. It just bloats the code and doesn't really do anything useful. Just use FlightServer.

Reply:
Just using FlightServer

Reply:
Sorry, this is the opposite of what I meant: we should demonstrate implementing FlightProducer, but we should not use anything from `org.apache.arrow.flight.example`, and we don't need to copy how it creates an "ExampleFlightServer" class. We should just directly create "FlightServer".

Reply:
I got it now, let me implement on that way